conda/build.sh

```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 set -e
 export ONNX_BINARY_BUILD=1
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    export MACOSX_DEPLOYMENT_TARGET=10.9
```

Comment:
I vaguely remember on my macbook I need to export this as "10.12"

Reply:
10.9 empirically worked on my macOS Sierra 10.12.6, so I am not sure what is going on here.

Reply:
Oh actually it was installing pytorch, not onnx.

Reply:
Could it be that you are affected by https://bugs.python.org/issue9516 ? This is inconsistent with the version number you reported which should have had this bug fixed. Could you post a longer trace about this issue?

Note that the semantics of `MACOSX_DEPLOYMENT_TARGET` is that it specifies the *minimum* Mac OS X version that we want to be able to run an executable on. So lower is better.

Reply:
Unfortunately the stack trace doesn't have any useful information:

```
$ MACOSX_DEPLOYMENT_TARGET=10.9 python setup.py develop
zip_safe flag not set; analyzing archive contents...

Installed /private/tmp/onnx/.eggs/pytest_runner-2.12.1-py2.7.egg
running develop
running create_version
running egg_info
creating onnx.egg-info
writing requirements to onnx.egg-info/requires.txt
writing onnx.egg-info/PKG-INFO
writing top-level names to onnx.egg-info/top_level.txt
writing dependency_links to onnx.egg-info/dependency_links.txt
writing manifest file 'onnx.egg-info/SOURCES.txt'
reading manifest file 'onnx.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'onnx.egg-info/SOURCES.txt'
running build_ext
running build_proto
compiling /private/tmp/onnx/onnx/onnx.proto
building 'onnx.onnx_cpp2py_export' extension
clang -fno-strict-aliasing -fno-common -dynamic -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/private/tmp/onnx -I/private/tmp/onnx/third_party/pybind11/include -I/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/include/python2.7 -I/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c /private/tmp/onnx/onnx/cpp2py_export.cc -o build/temp.macosx-10.12-x86_64-2.7/private/tmp/onnx/onnx/cpp2py_export.o -std=c++11 -stdlib=libc++
error: $MACOSX_DEPLOYMENT_TARGET mismatch: now "10.9" but "10.12" during configure
```

I suspect it's my python (which is from brew) was configured to "MACOSX_DEPLOYMENT_TARGET=10.12" when it was built.